xmodule/docs/decisions/0003-library-content-block-schema.rst

```diff
@@ -0,0 +1,201 @@
+
+   * - selected
+     - user_state, list of (block type, block id) tuples
+     - []
+     - [("problem", "A"), ("problem", "F")]
+     - An ordered sub-list of the children, to be shown to a given learner. Once populated, this is kept stable for the learner, unless a the children or max_count are changed, in which case it will have blocks added or removed as necessary. If the list changes, it is re-shuffled.
+
+
+Decision
+********
+
+We will introduce the support for a "manual selection mode", without explicitly making it *mode* as hinted at by the old block schema. This will lead to a cleaner and more flexible implementation. We will achieve this with two new boolean settings: **manual** and **shuffle**:
+
+* When manual is *disabled* and shuffle is *enabled*, the block will behave as it did before. That is, it will import the entire library (filtered by capa_type) as its children, and present each user a random subset based on max_count. For backwards compatibility, these will be the default value of the settings.
+
+* When manual is *enabled*, shuffle is *disabled*, and max_count is set to -1, the block will behave in the "static" mode. That is, the user will be prompted to select specific blocks in the library, and each user will be presented those blocks in order.
+
+* When manual is *enabled*, any filter fields (currently just capa_type, but perhaps more in the future) will be ignored for the purposes of CMS deciding which blocks to import. That's because in the event that manually-picked children clash with the filters, we need to decide who would "win", and we are deciding here that the manually-picked children would win. However, we could choose to have the block-picker filter blocks based on filters.
+
+We will also remove the **mode** field, as it is no longer needed.
+
+The interaction between manual, shuffle, and max_count yields a matrix of 8 different behaviors:
+
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+|                                     | **manual = False**                                            | **manual = True**                                               |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+| **shuffle = True,  max_count = -1** | Entire library in random order (V1's "random mode")           | Manual subset of library in random order                        |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+| **shuffle = True,  max_count > 0**  | Random subset of library in random order (V1's "random mode") | Random subset of manual subset of library in random order       |
```

Comment:
It took me a minute to grok this, and I think it might help if we differentiate in our language between the subset of content chosen for the course, and the subset of content chosen for the user.

Reply:
@ormsbee I agree. I would love to call "subset of content chosen for the course" to be the `selected` blocks, but unfortunately, that's already what the block calls the "subset of content chosen for the user".

How about:
* We call the "subset of content chosen for the course" the **included blocks**, implemented as the block children. These blocks can be manually-included or automatically-included. They can also be filtered by capa_type.
* We call the "subset of content chosen for the user" the **LMS-selected blocks**, implemented by the user-scope `selected` field. We can select all included blocks (max_count < 0) or selected a random subset of included blocks (max_count > 0). They can be shuffled or not.